cosmos/models/Workflow.py

```diff
@@ -560,10 +562,21 @@ def _process_finished_tasks(jobmanager):
     for task in jobmanager.get_finished_tasks():
         if task.NOOP or task.exit_status == 0:
             task.status = TaskStatus.successful
-            yield task
+        elif task.scheduling_error:
```

Comment:
Cool.  It's good that we're separating retrying due to bad error codes vs due to scheduling errors (since in our case the latter is what we want to retry), but what happens if scheduling errors occur indefinitely?  seems like we will keep retrying forever.  

I want to merge but I don't think we should keep this code deployed into production after the cluster is back to normal without being able to set $number_of_retries_due_to_scheduling_errors back to 0.  If the problem re-occurs, better for the jobs to fail so we know what's going on.  I'm fine with that just being the retry part of this code commented out, and ready to be commented back in should we need it if you don't want to go through the trouble of making it a real parameter of Cosmos.

If at some point we go to the cloud, especially if we use spot instances, we'll likely want some nuance around these different types of failures.

Thoughts?